### ISSUE TITLE
Wrong octave system-wide

### ISSUE THREAD

**Issue by aik099** [djipco/webmidi#42]
When I press `Middle C`, then `noteon` event reports it as `C3` while it should be `C4`.

I'm using event listening code from README.md:

```js
 // Listen for a 'note on' message on all channels
  input.addListener('noteon', "all",
    function (e) {
      console.log("Received 'noteon' message (" + e.note.name + e.note.octave + ").");
    }
  );
```

Same happens, the other way around: attempt to play `C3` actually plays `C4`:

```js
output.playNote('C3', 'all', {duration: 2000, velocity: 0.25});
```

Here are octaves shown by Synthesia and it clearly shows that when `C3` is reported by `noteon` event, then Synthesia (and other programs) display it as `C4`:

![2018-06-08_0853](https://user-images.githubusercontent.com/1277526/41141162-983e8294-6af9-11e8-8ced-1f70bd501154.png)

I'm suspecting an issue when in math, when decoding/encoding midi messages.

When I press 1st key on the keyboard (`A0`) it's reported as `A-1`, which according to README.md is a special code for stop playing note on all channels (see `output.stopNote("C-1");` in there).

**Comment by aik099** [comment]
Removing last `- 1` from `WebMidi.prototype.getOctave` method should fix octave reporting in events:

old: `return Math.floor(parseInt(number) / 12 - 1) - 1;`
new: `return Math.floor(parseInt(number) / 12 - 1);`

Examples:

* MIDI Code: 60 (Middle C); Result: Floor(60 / 12 - 1) = 4
* MIDI Code: 21 (A0): Result: Floor(21 / 12 - 1) = 0

**Comment by aik099** [comment]
Changing `octave + 2` into `octave + 1` in the `WebMidi.prototype.noteNameToNumber` method should fix the `playNote` method:

old: `var result = ((octave + 2) * 12) + semitones;`
new: `var result = ((octave + 1) * 12) + semitones;`

Examples:

* Note: C4 (note "C", octave: 4); MIDI Code: ((4 + 1) * 12) + 0 = 60
* Note: A0 (note "A", octave: 0): MIDI Code: ((0 + 1) * 12) + 9 = 21

**Comment by aik099** [comment]
I can send a PR with above described changes as well.

**Comment by aik099** [comment]
I've also found https://www.midi.org/forum/830-midi-octave-and-note-numbering-standard article where a lot confusion happens between fact what octave is `Middle C` is it `C3`, `C4`, `C5` or something else.

**Comment by francoisgeorgy** [comment]
Hi,

I always thought that middle C = C4 was a standard or at least a "de facto" standard. I'm quite surprised this is not really the case.

But, it seems that there is kind of a consensus toward that C4 value. So, for me this is the best candidate. I vote for it :-)

Sorry not to be able to bring more to the discussion.

**Comment by aik099** [comment]
@francoisgeorgy , maybe you can tweet about it. My twitter followers are developers mostly, so I'm not even trying 😉 to tweet about it.

**Comment by djipco** [comment]
> I always thought that middle C = C4 was a standard or at least a "de facto" standard. I'm quite surprised this is not really the case.

You would find many people to disagree with that statement. For example, various pieces of hardware and software treat MIDI note 60 (middle C) as C3. Some notable examples that come to mind are Yamaha's hardware, Apple's Logic, Steinberg's Cubase, etc. Also, many people were taught middle C = C3 in music school.

Given the variations in practices and opinions on this matter, I think the best course of action is to align with some sort of standard. In this case, the most obvious one seems to be [scientific pitch notation](https://en.wikipedia.org/wiki/Scientific_pitch_notation).

So, I'm going to change WebMidi.js to define middle C as C4. However, this change is going to be accompanied by a way to globally modify this behaviour.

**Comment by aik099** [comment]
I'll added a note about code in 5a0e0aea33f1da329306586d0b90eaa7729c8ddf commit directly, because you're not using a PR.

Will test shortly.

**Comment by aik099** [comment]
Tested, that behavior explained in code above `WebMidi.octaveOffset` setting works as described:

* the C4 becomes C5 if I use `+1` offset
* the C4 becomes C3 if I use `-1` offset

**Comment by djipco** [comment]
Fixed in [release 2.1](https://github.com/djipco/webmidi/releases/tag/2.1.0).
